### PR TITLE
Feature/on error http response

### PR DESCRIPTION
**Pull request by joe94**

### CODE REVIEW

File: auth/enforcer.go [hunk 1]
```diff
@@ -59,13 +60,13 @@ func providePrimaryBasculeEnforcer() fx.Option {
...
 				Target: func() bascule.Validator {
-					return bascule.CreateNonEmptyPrincipalCheck()
+					return bchecks.NonEmptyPrincipal()
 				},
 			},
 			fx.Annotated{
 				Name: "primary_bearer_validator_type",
 				Target: func() bascule.Validator {
-					return bascule.CreateValidTypeCheck([]string{"jwt"})
+					return bchecks.ValidType([]string{"jwt"})
```
[the remark this answers is not in the document]
Comment (kristinapathak) [reply auth/enforcer.go:69]: `basculechecks` is for all checks aka `bascule.Validator`s, which is why I moved those there. I didn't think about the required import 😬 but luckily anyone importing it is already dependent on the `bascule` module.

File: auth/constructor.go [hunk 2]
```diff
@@ -73,13 +79,21 @@ func providePrimaryBasculeConstructorOptions(apiBase string) fx.Option {
...
+			Group: primaryBasculeCOptionsName,
 			Target: func() basculehttp.COption {
 				return basculehttp.WithParseURLFunc(basculehttp.CreateRemovePrefixURLFunc("/"+apiBase+"/", basculehttp.DefaultParseURLFunc))
 			},
 		},
+		fx.Annotated{
+			Group: primaryBasculeCOptionsName,
+			Target: func(in primaryBasculeOnHTTPErrorResponseIn) basculehttp.COption {
+				if in.OnErrorHTTPResponse == nil {
+					return nil
```
Comment (kristinapathak) [inline auth/constructor.go:92]: Won't this add `nil` to the list? I'm not sure that bascule is equipped to handle that?

A solution might be to use the `arrange.IfSet()` function, but I'm not sure.

File: chrysom/client.go [hunk 3]
```diff
@@ -166,13 +165,15 @@ type observerConfig struct {
...
-func NewClient(config ClientConfig, logger func(ctx context.Context) bascule.Logger) (*Client, error) {
+func NewClient(config ClientConfig, getLogger func(ctx context.Context) log.Logger) (*Client, error) {
 	err := validateConfig(&config)
 	if err != nil {
 		return nil, err
 	}
-	if logger == nil {
-		logger = bascule.GetDefaultLoggerFunc
+	if getLogger == nil {
+		getLogger = func(ctx context.Context) log.Logger {
```
[the remark this answers is not in the document]
Comment (kristinapathak) [reply chrysom/client.go:174]: Since we're moving away from `log.Logger`, I'm not sure we should put in the work?

**Review by kristinapathak — APPROVED** [review]
Looks good! Just had one question in the constructor setup.

File: auth/constructor.go (hunk 2 above)
Comment (joe94) [reply auth/constructor.go:92]: Yeah, nil will be added to the list but it'll be filtered out before using the options here https://github.com/xmidt-org/argus/blob/4761625cca08a0683bb6a464c31e1a71145e2651/auth/constructor.go#L95

**Pull request merged**